File: framework/wcf/Extensibility/Security/DurableIssuedTokenProvider/CS/RSTRSTR/RequestSecurityTokenBase.cs
//-----------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//-----------------------------------------------------------------------------
using System;

using System.ServiceModel;
using System.ServiceModel.Channels;

namespace Microsoft.Samples.DurableIssuedTokenProvider
{
    public abstract class RequestSecurityTokenBase : BodyWriter
    {
        // private members
        private string m_context;
        private string m_tokenType;
        private int m_keySize;
        private EndpointAddress m_appliesTo;
        
        // Constructors
        protected RequestSecurityTokenBase() : this(String.Empty,String.Empty,0, null)
        {
        }

        protected RequestSecurityTokenBase(string context, string tokenType, int keySize, EndpointAddress appliesTo )
            : base(true)
        {
            this.m_context = context;
            this.m_tokenType = tokenType;
            this.m_keySize = keySize;
            this.m_appliesTo = appliesTo;
        }

        // public properties
        public string Context
        {
            get { return m_context; }
            set { m_context = value; }
        }

        public string TokenType 
        {
            get { return m_tokenType; }
            set { m_tokenType = value; }
        }

        public int KeySize
        {
            get { return m_keySize; }
            set { m_keySize = value; }
        }

        public EndpointAddress AppliesTo
        {
            get { return m_appliesTo; }
            set { m_appliesTo = value; }
        }
    }
}

File: framework/wcf/Basic/Binding/WS/WS2007FederationHttp/VB/RSTRSTR/RequestSecurityTokenBase.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

Imports System

Imports System.ServiceModel
Imports System.ServiceModel.Channels

Namespace Microsoft.Samples.WS2007FederationHttpBinding
    Public MustInherit Class RequestSecurityTokenBase
        Inherits BodyWriter
        ' private members
        Private m_context As String
        Private m_tokenType As String
        Private m_keySize As Integer
        Private m_appliesTo As EndpointAddress

        ' Constructors
        Protected Sub New()
            Me.New([String].Empty, [String].Empty, 0, Nothing)
        End Sub

        Protected Sub New(ByVal context As String, ByVal tokenType As String, ByVal keySize As Integer, ByVal appliesTo As EndpointAddress)
            MyBase.New(True)
            Me.m_context = context
            Me.m_tokenType = tokenType
            Me.m_keySize = keySize
            Me.m_appliesTo = appliesTo
        End Sub

        ' public properties
        Public Property Context() As String
            Get
                Return m_context
            End Get
            Set(ByVal value As String)
                m_context = value
            End Set
        End Property

        Public Property TokenType() As String
            Get
                Return m_tokenType
            End Get
            Set(ByVal value As String)
                m_tokenType = value
            End Set
        End Property

        Public Property KeySize() As Integer
            Get
                Return m_keySize
            End Get
            Set(ByVal value As Integer)
                m_keySize = value
            End Set
        End Property

        Public Property AppliesTo() As EndpointAddress
            Get
                Return m_appliesTo
            End Get
            Set(ByVal value As EndpointAddress)
                m_appliesTo = value
            End Set
        End Property
    End Class
End Namespace
